allennlp/nn/util.py

```diff
@@ -184,7 +187,9 @@ def masked_log_softmax(vector, mask):
     return torch.nn.functional.log_softmax(vector)
 
 
-def viterbi_decode(tag_sequence: torch.Tensor, transition_matrix: torch.Tensor):
+def viterbi_decode(tag_sequence: torch.Tensor,
+                   transition_matrix: torch.Tensor,
+                   tag_evidence: Optional[List[int]] = None):
```

Comment:
When I saw the title of this PR in the email, I wondered what "evidence" was.  If you had instead called this "observed_tags", I would have immediately known what you were doing.  I _think_ "observed" is the more standard term for what you're doing here.  Your docstring below seems to also agree with me.